Demo/Common/IntegrationExamplePartialPayment.swift

```diff
@@ -0,0 +1,111 @@
+            switch self {
+            case .nonBalance:
+                return "No Balance"
+            }
+        }
+    }
+
+    internal func checkBalance(_ data: PaymentComponentData,
+                               from component: PaymentComponent,
+                               completion: @escaping (Result<Balance, Error>) -> Void) {
+        let balance = Balance(availableAmount: Payment.Amount(value: 10000, currencyCode: "CAD"),
+                              transactionLimit: Payment.Amount(value: 100000, currencyCode: "CAD"))
+        completion(.success(balance))
+//        let request = BalanceCheckRequest(data: data)
+//        apiClient.perform(request) { [weak self] result in
+//            self?.handle(result: result, completion: completion)
+//        }
+    }
+
+    private func handle(result: Result<BalanceCheckResponse, Error>,
+                        completion: @escaping (Result<Balance, Error>) -> Void) {
+        switch result {
+        case let .success(response):
+            handle(response: response, completion: completion)
+        case let .failure(error):
+            handle(error: error, completion: completion)
+        }
+    }
+
+    private func handle(response: BalanceCheckResponse, completion: @escaping (Result<Balance, Error>) -> Void) {
+        guard let availableAmount = response.balance else {
+            finish(with: GiftCardError.nonBalance)
+            completion(.failure(GiftCardError.nonBalance))
+            return
+        }
+        let balance = Balance(availableAmount: availableAmount, transactionLimit: response.transactionLimit)
+        completion(.success(balance))
+    }
+
+    private func handle(error: Error, completion: @escaping (Result<Balance, Error>) -> Void) {
+        finish(with: error)
+        completion(.failure(error))
+    }
+
+    internal func requestOrder(_ data: PaymentComponentData,
+                               from component: PaymentComponent,
+                               completion: @escaping (Result<PartialPaymentOrder, Error>) -> Void) {
+        let result = PartialPaymentOrder(pspReference: "ef2ef", orderData: "ef2f23f", reference: "e2f2f2", remainingAmount: .init(value: 7408, currencyCode: "CAD"), expiresAt: nil)
+        completion(.success(result))
+//        let request = CreateOrderRequest(amount: payment.amount, reference: UUID().uuidString)
+//        apiClient.perform(request) { [weak self] result in
+//            self?.handle(result: result, completion: completion)
+//        }
+    }
+
+    private func handle(result: Result<CreateOrderResponse, Error>,
+                        completion: @escaping (Result<PartialPaymentOrder, Error>) -> Void) {
+        switch result {
+        case let .success(response):
+            self.order = response.order
+            completion(.success(response.order))
+        case let .failure(error):
+            finish(with: error)
+            completion(.failure(error))
```

Comment:
looks like we are calling 2 completion handlers here

Reply:
finish is not a completion handler, right? its business logic of the merchant to show an error alert or something